drivers/1wire/1wire.c

```diff
@@ -554,6 +562,7 @@ int onewire_search(FAR struct onewire_master_s *master,
           nslaves_match++;
         }
 
+      onewire_sem_post(master);
       return (ret < 0) ? ret : nslaves_match;
```

Comment:
remove line 565 and change line 566 to:
goto err_unlock;

Reply:
I will rename err_unlock to unlock because the case in line 565 isn't explicit an error.